app/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertRepository.java

```diff
@@ -0,0 +1,365 @@
+        _alertQueryGenerator = alertQueryGenerator;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void open() {
+        assertIsOpen(false);
+        LOGGER.debug().setMessage("Opening alert repository").log();
+        _isOpen.set(true);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void close() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Closing alert repository").log();
+        _isOpen.set(false);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<Alert> get(final UUID identifier) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Getting alert")
+                .addData("alertId", identifier)
+                .log();
+
+        final models.ebean.Alert ebeanAlert = Ebean.find(models.ebean.Alert.class)
+                .where()
+                .eq("uuid", identifier)
+                .findUnique();
+        if (ebeanAlert == null) {
+            return Optional.empty();
+        }
+        final Alert result = new DefaultAlert.Builder()
+                .setId(ebeanAlert.getUuid())
+                .setName(ebeanAlert.getName())
+                .setContext(ebeanAlert.getContext())
+                .setCluster(ebeanAlert.getCluster())
+                .setService(ebeanAlert.getService())
+                .setMetric(ebeanAlert.getMetric())
+                .setStatistic(ebeanAlert.getStatistic())
+                .setPeriod(Period.seconds(ebeanAlert.getPeriod()))
+                .setOperator(ebeanAlert.getOperator())
+                .setValue(new DefaultQuantity.Builder()
+                        .setValue(ebeanAlert.getQuantityValue())
+                        .setUnit(ebeanAlert.getQuantityUnit())
+                        .build())
+                .setExtensions(mergeExtensions(ebeanAlert.getNagiosExtension()))
+                .build();
+        return Optional.of(result);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AlertQuery createQuery() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Preparing query").log();
+        return new DefaultAlertQuery(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public QueryResult<Alert> query(final AlertQuery query) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Querying")
+                .addData("query", query)
+                .log();
+
+        // Create the base query
+        final PagedList<models.ebean.Alert> pagedAlerts = _alertQueryGenerator.createAlertQuery(query);
+
+        // Compute the etag
+        final SqlQuery sqlQuery = Ebean.createSqlQuery("SELECT CURRVAL('portal.alerts_etag_seq') AS etag;");
+        final Long etag = sqlQuery.findUnique().getLong("etag");
+
+        final List<Alert> values = new ArrayList<>();
+        pagedAlerts.getList().forEach(ebeanAlert -> {
```

Comment:
This building code is duplicated from above. Perhaps using a private conversion method and function reference would be the way to go here.